transport/http/client.go

```diff
@@ -118,6 +140,13 @@ func (c Client) Endpoint() endpoint.Endpoint {
 	}
 }
 
+// ClientFinalizerFunc can be used to perform work at the end of a client HTTP
+// request, after the response is returned. The principal
+// intended use is for request logging. In addition to the response code
+// provided in the function signature, additional response parameters are
+// provided in the context under keys with the ContextKeyResponse prefix.
+type ClientFinalizerFunc func(ctx context.Context, code int, r *http.Request)
```

Comment:
Hmm. This is the same signature as in the ServerFinalizer, but I'm not sure it actually makes sense here. In the server we're guaranteed to have an http.Request, and we're guaranteed to have a code, so they make sense as parameters. But in the client, we may not make it all the way to generating an http.Request, and we may not get a status code back. And, we have an explicit error value that will always be returned via the endpoint (though it may be nil). So I think the better signature here is

```go
type ClientFinalizerFunc func(ctx context.Context, err error) 
```

What do you think?